airbyte-integrations/connectors/source-http-request/source_http_request/source.py

```diff
@@ -62,25 +62,42 @@ def discover(self, logger: AirbyteLogger, config: json) -> AirbyteCatalog:
         }
 
         # json body will be returned as the "data" stream". we can't know its schema ahead of time, so we assume it's object (i.e. valid json).
-        return AirbyteCatalog(streams=[AirbyteStream(name=SourceHttpRequest.STREAM_NAME, json_schema=json_schema)])
+        return AirbyteCatalog(
+            streams=[
+                AirbyteStream(
+                    name=SourceHttpRequest.STREAM_NAME, json_schema=json_schema
+                )
+            ]
+        )
 
     def read(
-        self, logger: AirbyteLogger, config: json, catalog: ConfiguredAirbyteCatalog, state: Dict[str, any]
+        self,
+        logger: AirbyteLogger,
+        config: json,
+        catalog: ConfiguredAirbyteCatalog,
+        state: Dict[str, any],
     ) -> Generator[AirbyteMessage, None, None]:
         r = self._make_request(config)
         if r.status_code != 200:
             raise Exception(f"Request failed. {r.text}")
 
         # need to eagerly fetch the json.
         data = r.json()
+
+        if isinstance(data, list) and isinstance(data[0], list):
```

Comment:
we should also verify `len(data) > 0`

Reply:
hey @sherifnada, I'm actually going to close this PR
I agree with @marcosmarxm that the responses from the US Census API are so unconventional that they would be better supported to a separate "Source"